src/stdlib_linalg.fypp

```diff
@@ -214,6 +219,25 @@ module stdlib_linalg
     #:endfor
   end interface is_hessenberg
 
+  #:if defined("USE_OPENBLAS")
+  interface solve
+    !! version: experimental
+    !!
+    !! Solve a linear matrix equation, or system of linear scalar equations
+    !! ([Specification](../page/specs/stdlib_linalg.html#
+    !! solve-solve-a-linear-matrix-equation))
+    #:set REAL_KINDS_TYPES_GESV = list(zip(BLAS_REAL_KINDS, REAL_TYPES, BLAS_REAL_GESV))
+    #:set CMPLX_KINDS_TYPES_GESV = list(zip(BLAS_CMPLX_KINDS, CMPLX_TYPES, BLAS_CMPLX_GESV))
+    #:set RC_KINDS_TYPES_GESV = REAL_KINDS_TYPES_GESV + CMPLX_KINDS_TYPES_GESV
+    #:for k1, t1, g1 in RC_KINDS_TYPES_GESV
+    module function solve_${t1[0]}$${k1}$(a, b) result(x)
```

Comment:
I would prefer a subroutine-style interface, e.g.
```fortran
subroutine solve(X, A, B)
```

Reply:
I agree with the above. It makes more sense to use a subroutine, so that the code stays performant when large matrices are involved. 

Also, would it make sense to call the procedure something like `linsolve`, instead of just `solve`, anticipating that in the future a nonlinear solver will be added as well?